ppl/zqd/apiserver/filedb.go

```diff
@@ -0,0 +1,201 @@
+package apiserver
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"sync"
+
+	"github.com/brimsec/zq/api"
+	"github.com/brimsec/zq/pkg/iosrc"
+	"github.com/brimsec/zq/zqe"
+)
+
+type FileDb struct {
+	mu   sync.Mutex
+	path iosrc.URI
+}
+
+func CreateFileDb(ctx context.Context, path iosrc.URI, rows []SpaceRow) (*FileDb, error) {
+	ldb := &FileDb{path: path}
+	if err := ldb.save(ctx, rows); err != nil {
+		return nil, err
+	}
+	return ldb, nil
+}
+
+func OpenFileDb(ctx context.Context, path iosrc.URI) (*FileDb, error) {
+	ldb := &FileDb{path: path}
+	// Verify file exists & is readable.
+	if _, err := ldb.load(ctx); err != nil {
+		return nil, err
+	}
+	return ldb, nil
+}
+
+const dbversion = 4
+
+type dbdataV4 struct {
+	Version   int        `json:"version"`
+	SpaceRows []SpaceRow `json:"space_rows"`
+}
+
+func (ldb *FileDb) load(ctx context.Context) ([]SpaceRow, error) {
+	b, err := iosrc.ReadFile(ctx, ldb.path)
+	if err != nil {
+		return nil, err
+	}
+	var lf dbdataV4
+	if err := json.Unmarshal(b, &lf); err != nil {
+		return nil, err
+	}
+	return lf.SpaceRows, nil
```

Comment:
Check the version?

Reply:
done in https://github.com/brimsec/zq/pull/1693/commits/5974423b48c726319a9414cb575a93e9f21c5785